src/main/java/com/obj/nc/repositories/DeliveryInfoRepository.java

```diff
@@ -154,29 +154,34 @@ long countByMessageIdAndStatus(@Param("messageId") UUID messageId,
 
     long countByEndpointIdAndProcessedOnAfter(UUID endpointId, Instant timestamp);
 
-    @Query(value = "select " +
-            "    di.id AS delivery_id, " +
-            "    m.id AS message_id, " +
-            "    e.endpoint_type AS endpoint_type, " +
-            "    di.status AS delivery_status, " +
-            "    di.additional_information AS additional_information, " +
-            "    m.reference_number AS reference_number " +
-            "from nc_delivery_info di " +
-            "join nc_message m " +
-            "on di.message_id = m.id " +
-            "join nc_endpoint e " +
-            "on di.endpoint_id = e.id " +
-            "where di.status IN ('SENT', 'DELIVERY_PENDING') " +
-            "and e.endpoint_type IN (:endpointTypes) " +
-            "and di.processed_on > :timestamp " +
-            "order by di.processed_on",
-            rowMapperClass = DeliveryInfoDtoMapper.class)
-    List<DeliveryInfoDto> findUnfinishedDeliveriesNotOlderThan(@Param("timestamp") Instant timestamp, @Param("endpointTypes") List<String> endpointTypes);
-
-    @Modifying
-    @Query(value = "update nc_delivery_info " +
-            "set status = :status, " +
-            "    additional_information = :additional_information " +
-            "where id = :id ")
-    void updateStatusAndAdditionalInformationById(@Param("status") DELIVERY_STATUS status, @Param("additional_information") String additionalInformation, @Param("id") UUID id);
+    @Query(
+        value = "select distinct on (m.id)" +
+                "    di.id AS delivery_id, " +
+                "    m.id AS message_id, " +
+                "    e.endpoint_type AS endpoint_type, " +
+                "    di.status AS delivery_status, " +
+                "    di.additional_information AS additional_information, " +
+                "    m.reference_number AS reference_number " +
+                "from nc_delivery_info di " +
+                "join nc_message m " +
+                "on di.message_id = m.id " +
+                "join nc_endpoint e " +
+                "on di.endpoint_id = e.id " +
+                "where di.status IN ('SENT', 'DELIVERY_PENDING') " +
+                "and e.endpoint_type IN (:endpointTypes) " +
+                "and di.processed_on > :timestamp " +
+                "and not exists ( " +
```

Comment:
skus pls. pozret, ci toto nevieme vyriesit cez ten di.status in (..)

Reply:
@cuzyjan 
- tato metoda findUnfinishedDeliveriesNotOlderThan(...) vracia vsetky neterminalne deliveryInfos sprav. tj. keby tam nebola cast "and not exists...", mohla by existovat sprava, ktora bola poslana a uz aj dorucena - robime insert, takze existuju obidve deliveryInfos a pytali by sme sa na stav spravy o ktorej uz vieme, ze bola dorucena, musime preto odfiltrovat spravy, ktore uz maju terminalne deliveryInfo

- PROCESSING nie je pri ('SENT', 'DELIVERY_PENDING') preto, ze ked sme este neposlali spravu do GAPu (/mailservera,..) tak nema vyznam sa ho pytat na jej stav

Reply:
ok, jasne,.. pls. dopis tam comment aby sme to niekedy nezmazali. 
mozme mergnut